source/AlertLabel.cpp

```diff
@@ -0,0 +1,74 @@
+/* AlertLabel.cpp
+Copyright (c) 2022 by Daniel Yoon
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "AlertLabel.h"
+
+#include "Angle.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "GameData.h"
+#include "Government.h"
+#include "LineShader.h"
+#include "Projectile.h"
+#include "pi.h"
+#include "PointerShader.h"
+#include "RingShader.h"
+#include "Ship.h"
+#include "Sprite.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+namespace {
+	const double DANGEROUS_ABOVE = .05;
+}
+
+
+
+AlertLabel::AlertLabel(const Point position, const Projectile &projectile, const shared_ptr<Ship> flagship, const double zoom)
+		: position(position * zoom), zoom(zoom)
+{
+	isTargetingMe = projectile.TargetPtr() == flagship;
+	double maxHP = flagship->Attributes().Get("hull") + flagship->Attributes().Get("shield");
+	double missileDamage = projectile.GetWeapon().HullDamage() + projectile.GetWeapon().ShieldDamage();
+	damagePercent = missileDamage / maxHP;
+	isDangerous = damagePercent > DANGEROUS_ABOVE;
+
+	if(!isTargetingMe)
```

Comment:
Switch the condition around, i.e. `if(isTargetingMe)` because you have an else anyways. You can also do an if else chain if you move the isDangerouns `if` first.